### PR TITLE
Fix: url with anchors are being replaced with url without anchors

### LINKED ISSUE #283: url with anchors are being replaced with url without anchors
Eg: go to http://putaindecode.io/en/articles/js/react/unit-testing/, click on a title anchor. Link works, but anchor is quickly stripped from the url bar

### PR DESCRIPTION
**Pull request by thangngoc89**
Close #283

### CODE REVIEW

**Comment by MoOx** [comment]
I would like to cover this with test this time :)

File: src/PageContainer/component.js [hunk 1]
```diff
@@ -100,8 +100,10 @@ export default class PageContainer extends Component {
     }
 
     const item = find(context.collection, pageUrl)
-
+    console.log(isClient)
+    console.log(typeof window !== "undefined")
```
Comment (thangngoc89) [inline src/PageContainer/component.js]: I injected window to global var

@MoOx any idea why 

```
console.log(isClient) --> false
console.log(typeof window !== "undefined") --> true
```

Comment (MoOx) [inline src/PageContainer/component.js]: You define window after importing the file, and isClient is defined at the top of the file.

Comment (thangngoc89) [inline src/PageContainer/component.js]: Got it. So change isClient to a function then ?

anyway, I need to pull babel-plugin-rewire browserHistory. So this PR is blocked by #286

Comment (MoOx) [inline src/PageContainer/component.js]: Or maybe refactor the code to pass this as props.
And isClient can be replaced by its value (simpler to test)

Comment (thangngoc89) [inline src/PageContainer/component.js]: Yeah, that's easier. Are you ok with pulling babel-plugin-rewire ? Do you have any refer method ?

Comment (MoOx) [inline src/PageContainer/component.js]: Looks like a bad design to need this thing.
What about finding a way to pass this as a prop?

Comment (MoOx) [inline src/PageContainer/component.js]: (Or via the context)

Comment (thangngoc89) [inline src/PageContainer/component.js]: We only use browserHistory in PageContainer so no need for context. So browserHistory, isClient, ... will be passed via props

Comment (MoOx) [inline src/PageContainer/component.js]: passed via props means that something is gonna pass those thing.
isClient is not really useful, just drop it an use a simple test directly in the component.

Comment (thangngoc89) [inline src/PageContainer/component.js]: We have react-redux connect to pass those via props.

So using `typeof window !== 'undefined'` directly right ?

Comment (MoOx) [inline src/PageContainer/component.js]: Redux will be removed one day or another :)

Comment (thangngoc89) [inline src/PageContainer/component.js]: Ouch. browserHistory required may DOM API, can't mock them all

Comment (MoOx) [inline src/PageContainer/component.js]: We will need jsdom one day or another. We should add it now.

Comment (MoOx) [inline src/PageContainer/component.js]: First, double check that jsdom support browser history :/

Comment (thangngoc89) [inline src/PageContainer/component.js]: On it

**Comment by thangngoc89** [comment]
Ouch. jsdom doesn't work with `history` package. `history` requires `window.sessionStorage` which jsdom [doesn't have](https://github.com/tmpvar/jsdom/issues/1137)

**Comment by MoOx** [comment]
What about mocking window requirements?!

**Comment by thangngoc89** [comment]
I'm mocking it. But history package uses a large browser API. window.sessionStorage wasn't the only thing it required.

```
SecurityError
    HistoryImpl._sharedPushAndReplaceState (/home/khoa/web/statinamic/node_modules/jsdom/lib/jsdom/living/window/History-impl.js:87:15)
    HistoryImpl.replaceState (/home/khoa/web/statinamic/node_modules/jsdom/lib/jsdom/living/window/History-impl.js:72:10)
    History.replaceState (/home/khoa/web/statinamic/node_modules/jsdom/lib/jsdom/living/generated/History.js:89:58)
```

**Comment by MoOx** [comment]
So maybe mocking browser-history like your proposed before...?

**Comment by thangngoc89** [comment]
I managed to make it work with jsdom :+1: . Setup is complete. Now it is the real testing part

**Comment by MoOx** [comment]
This is reallyyyy nice. Can you just add the interesting test ? :D

**Comment by thangngoc89** [comment]
@MoOx  done

File: src/PageContainer/component.js [hunk 2]
```diff
@@ -6,14 +6,14 @@ import urlify from "../_utils/urlify"
...
+const isClient = () => typeof window !== "undefined"
+const isDevelopmentClient = () => isDevelopment() && isClient
 
 let catchLinks
 let browserHistory
 
-if (isClient) {
+if (isClient()) {
   catchLinks = require("../_utils/catch-links").default
   browserHistory = require("../client").browserHistory
```
Comment (thangngoc89) [inline src/PageContainer/component.js]: @MoOx here is why I need to clean node cache. I need to refresh `browserHistory`

Comment (MoOx) [inline src/PageContainer/component.js]: oh, ok.

**Comment by MoOx** [comment]
I rebase this ;)

**Comment by thangngoc89** [comment]
Thanks

**Pull request merged**